README.md

```diff
@@ -251,7 +251,22 @@ Example run customization:
                 "labels": {
                     "label1": "value1",
                     "label2": "value2"
-                }
+                },
+                "ports": [
+                    "80:80",
```

Comment:
I wonder if it'd be better to split this port mapping string (as well as the external hosts string below) into a separate object, like is done for volumes.  The string format is tied to the Docker CLI (which could change, though unlikely) and may not be the most straightforward to a new-to-Docker user.  It's also harder to (statically) validate and provide IntelliSense for.

Maybe something like this:

```json
"ports": [
  {
    "containerPort": 80,
    "localPort": 80
  },
  {
    "containerPort": 443,
    "localPort": 443
  }
]
```

It's admittedly more verbose, but the user's unlikely edit these configurations often.  It seems better to aim for clarity than terseness.

Reply:
I see your point here and I think the object way it's more accurate so it can provide intelliSense for it and abstract from the docker related form, so the user does not need to know hoy to write the exact string.

For the ports I'll provide an object like the following:

```json
"ports": [
  {
    "hostPort": 80,
    "containerPort": 80
  },
  {
    "containerPort": 443
  },
  {
    "containerPort": 6029,
    "protocol": "udp"
  },
  {
    "containerPort": 6029,
    "protocol": "tcp"
  }
]
```
Alongside this object I'll check if ports have been configured so it does not execute with the `-P` flag, and only ports indicated will be mapped.

With this, you can provide if you want to allocate an exact port for the container port associated or just want it to get randomized, also you can just expose a port from an specified protocol `(tcp|udp)`.

For the external hosts:

I'll change the name to `extraHosts` to keep consinstence with the docker compose label as here we are adding an "extra host" under the `--add-host` instruction.

The final object would look like this:

```json
"extraHosts": [
  {
    "hostname": "some-hostname",
    "ip": "some-ip"
  },
  {
    "hostname": "some-other-hostname",
    "ip": "some-other-ip"
  }
]
```